LayoutKit/LayoutArrangement.swift

```diff
@@ -150,9 +150,6 @@ extension View {
      It doesn't do anything of view is already a subview.
      */
     fileprivate func addSubview(_ view: View, maintainCoordinates: Bool) {
-        if view.superview == self {
```

Comment:
The bug is described a bit in #66, but the tl;dr is that repeated applications of a layout to the same view hierarchy wouldn't guarantee the order of the subviews when some are reused and others are not.

The code that I removed was added as a unverified performance optimization (the internet thinks that addSubviews is slow). Unfortunately, this short circuit doesn't result in the correct ordering of the subviews (addSubview always pulls the subview to the top).

So, we can either rely on addSubviews to do the right thing (this PR) or an alternate implementation would look like this:

```swift
fileprivate func addSubview(_ view: View, maintainCoordinates: Bool) {
    if view.superview == self {
        bringSubview(toFront: view)
        return
    }
    if maintainCoordinates {
        let frame = view.convertToAbsoluteCoordinates(view.frame)
        addSubview(view)
        view.frame = view.convertFromAbsoluteCoordinates(frame)
    } else {
        addSubview(view)
    }
}
```

I am somewhat indifferent between these so I chose the "less code" option. I want to do a perf test to see if the "more code" option actually performs better.

Thoughts?